### ISSUE TITLE
[Clojure] Add support for keywords

### ISSUE THREAD

**Issue by hanyuone** [sublimehq/Packages#795]
The syntax highlighter for Clojure doesn't support the valid highlighting of keywords, which are basically strings with a colon at the front:

```clojure
:foo ; This is a keyword
```

Currently, though, Sublime highlights keywords like so (in Monokai):
![clojure](https://cloud.githubusercontent.com/assets/14177193/22503763/5a32711e-e8c7-11e6-93f7-52e6d07cd9d3.PNG)

Code:

```clojure
(defn change-disp []
  [:div.change-disp
   [:input {:type "button"
            :value "Display stuff"
            :on-click #(swap! display? not)}]])
```

The colons are highlighted (presumably because it's punctuation), but the rest isn't highlighted.

**Comment by djspiewak** [comment]
For context, clojure keywords should be scoped the same as Ruby/Scala symbols, since their usage is almost equivalent.

**Comment by Qwerp-Derp** [comment]
Also, functions with slashes in them (functions from other Clojure files) aren't highlighted at all - only the `/` is highlighted:

![clojure2](https://cloud.githubusercontent.com/assets/14177193/22544082/71f39cb8-e987-11e6-9d5b-5f6967682af3.png)

(This also demonstrates the first issue.)

**Comment by FichteFoll** [comment]
Please include the text used in screenshots for easier reproduction via copy and paste.

**Comment by Qwerp-Derp** [comment]
@FichteFoll I included that.

**Comment by Qwerp-Derp** [comment]
Bump: anything useful yet?

**Comment by keith-hall** [comment]
Seems like you could just modify your color scheme to highlight `source.symbol.clojure`, no?

**Comment by Qwerp-Derp** [comment]
@keith-hall That did (partially) solve the issue for the highlighter, but the `:` was still being highlighted incorrectly as punctuation. I believe this might be a "bug" (maybe not) with the actual Clojure syntax thing itself.

Picture and code for reference:

![clojure](https://cloud.githubusercontent.com/assets/14177193/23158507/903fcc0a-f873-11e6-9dbc-a9a2c233fc63.PNG)

```clojure
(defproject experiments "0.1.0-SNAPSHOT"
  :description "FIXME: write description"
  :url "http://example.com/FIXME"
  :license {:name "Eclipse Public License"
            :url "http://www.eclipse.org/legal/epl-v10.html"}
            
  :dependencies [[org.clojure/clojure "1.8.0"]]
  :main experiments.core/-main)
```

**Comment by keith-hall** [comment]
I guess you could work around it by also highlighting `keyword.operator.symbol.clojure` the same as `source.symbol.clojure`. 

I guess the syntax def could scope the `:` as `source.symbol.clojure` and replace the `keyword.operator.symbol.clojure` scope with a `punctuation` scope, as most color schemes won't target these I think. I'll leave that for the experts to discuss/decide, I'm just spouting out ideas :)

**Comment by djspiewak** [comment]
Honestly it's probably easiest just to modify the mode to scope things appropriately. The Clojure mode is still pretty simple since it hasn't been updated to the new stateful hotness, so it should be almost as easy as tweaking a regex. That also has the advantage of being something that is pull requestable.

**Comment by deathaxe** [comment]
@wbond seems fixed by #1259

**Comment by wbond** [comment]
This issue has been solved by a previous commit, hence it is being closed.
